### PR TITLE
early WIP, base structure of the gatsby-source-sql plugin

### PR DESCRIPTION
**Pull request by daubaris**
#8714 This should have status: WIP. Seeking for a helping hand.

Hello, so I've made some progress with the plugin. What can it do right now:
- Connect to mysql,
- Accept queries and execute them.
- Create nodes. (You can see them in GraphiQL under content field.)

The result of the query is always a string and looks like this (depending on the query ofc):
`"content": "[{\"name\":\"paul\"},{\"name\":\"alice\"},{\"name\":\"tom\"},{\"name\":\"junior\"}]"`
and I'm not quite sure if that's correct way of outputting it even though in mongodb plugin `mongodbCloudDocuments` internal content displays data the same way.

The second thing is that I cant really figure out a way on how to query the data from mysql using GraphQL, which by the way mongodb plugin can.

Could anyone direct me to the right path?

### CODE REVIEW

**Comment by pieh** [comment]
> The second thing is that I cant really figure out a way on how to query the data from mysql using GraphQL, which by the way mongodb plugin can.
> 
> Could anyone direct me to the right path?

Just looking at code this should work. Gatsby uses `node.internal.type` so in case of this code it would ``type: `Queries` `` - there is no `queries`/`allQueries` root fields in your graphql schema after you use your plugin?

**Comment by LekoArts** [comment]
@daubaris Hi 👋 
Do have time (and also desire) to work on this PR? We could help you finish it by answering your questions.
There is also:
https://www.gatsbyjs.org/docs/create-source-plugin/
https://www.gatsbyjs.org/docs/source-plugin-tutorial/

It would also help us if you could setup a demo endpoint we could query, too.

File: packages/gatsby-source-sql/package.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "gatsby-source-sql",
```
Comment (wardpeet) [inline packages/gatsby-source-sql/package.json]: you should rename it to be a vendor specific package: `gatsby-source-mysql` there are a lot of sql variations and you don't want them all in one package

**Comment by LekoArts** [comment]
Hey! Thanks so much for opening this pull request!

Since another user did finish this PR ([Link to discussion](https://github.com/gatsbyjs/gatsby/issues/8714#issuecomment-452944867)), I'll close your PR in favor of it.

We absolutely want to have you as a contributor, so please take a look at [our open issues][open-issues] for ideas, and please reach out to us on Twitter at `@gatsbyjs` with questions.

We offer pair programming sessions if you’d like to work with one of our maintainers to make your first contribution.

Thanks again, and we look forward to seeing more PRs from you in the future! :muscle::purple_heart:

[open-issues]: https://github.com/gatsbyjs/gatsby/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc

**Pull request closed without merging**